File: frontend/src/types/materialLedger.ts
export type { ImportResult, PageResult } from '@/types/common'

export interface MaterialLedger {
  id: number
  category: string
  genericName: string
  brand: string | null
  name: string
  model: string
  binLocation: string
  stockQuantity: number
  unitPrice: number | null
  remark: string | null
  createdAt?: string
  updatedAt?: string
}

export interface MaterialSavePayload {
  category: string
  genericName: string
  brand?: string | null
  name: string
  model: string
  binLocation: string
  unitPrice?: number | null
  remark?: string | null
}

export interface MaterialQuery {
  category?: string
  genericName?: string
  brand?: string
  name?: string
  model?: string
  binLocation?: string
  ids?: number[]
  page?: number
  pageSize?: number
}

export interface FilterLinkageQuery {
  category?: string
  genericName?: string
  brand?: string
}

export interface FilterOptions {
  categories: string[]
  genericNames: string[]
  brands: string[]
  models: string[]
  binLocations: string[]
}

export interface BomCatalogItem {
  id: number
  category: string
  genericName: string
  brand: string | null
  name: string
  displayLabel: string
}

File: backend/src/main/java/com/storage/warehouse/entity/MaterialLedger.java
package com.storage.warehouse.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("material_ledger")
public class MaterialLedger {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String category;

    private String genericName;

    private String brand = "";

    private String name;

    private String model;

    private String binLocation;

    private Integer stockQuantity;

    private BigDecimal unitPrice;

    private String remark;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}
